protocols/relay/src/behaviour/handler.rs

```diff
@@ -716,6 +739,30 @@ impl ConnectionHandler for Handler {
             }
         }
 
+        // Process inbound requests
+        if let Poll::Ready(Some(result)) = self.inbound_requests_futs.poll_next_unpin(cx) {
+            let event = match result {
+                Ok(event) => ConnectionHandlerEvent::NotifyBehaviour(event),
+                Err(err) => {
+                    ConnectionHandlerEvent::Close(StreamUpgradeError::Apply(Either::Left(err)))
+                }
+            };
```

Comment:
Instead of pattern matching here, you could directly construct the correct `ConnectionHandlerEvent` within the future, then you could directly `return` the `Output` of the futures here.

Reply:
If you do that, you could even merge both `FuturesUnordered` into one and call them `worker_streams`.